### PR TITLE
Migrate subnets/vpc, kms key, multithreading validators

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
Moreover, commit 1 will: 
remove "AUTO" defaults,
rename simultaneous_multithreading to disable_simultaneous_multithreading, 
delete validators.py to symbolize the end of validators migration
fix a redundant patch in test_pcluster_configure

commit 2 will remove multi-inheritance of SharedEbs 

commit 3 will remove _add_validators logic and priority of validators. 
Instead of adding the validators into a list in the resource, we execute the validators directly when `validate()` is called.
Because we are executing the validators directly, the execution order is consistent with the order of the code. Therefore, `priority` is no longer necessary.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 1]
```diff
@@ -229,6 +229,10 @@ def validate_backup_id_unsupported_parameters(self, data, **kwargs):
 class SharedStorageSchema(BaseSchema):
     """Represent the generic SharedStorage schema."""
 
+    # FIXME: pcluster2 doesn't allow "^/?NONE$" mount dir to avoid an ambiguity in cookbook.
```
Comment (enrico-usai) [inline cli/src/pcluster/schemas/cluster_schema.py:232]: I'd add this check in the SharedDirValidator, to merge a working code.

File: cli/src/pcluster/models/cluster.py [hunk 2]
```diff
@@ -641,16 +663,34 @@ class BaseComputeResource(Resource):
...
         self._add_validator(NameValidator, name=self.name)
+        self._add_validator(
+            DisableSimultaneousMultithreadingArchitectureValidator,
+            disable_simultaneous_multithreading=self.disable_simultaneous_multithreading,
+            architecture=self.architecture,
+        )
+
+    @property
+    def architecture(self):
+        """Compute cluster's architecture based on its head node instance type."""
```
Comment (enrico-usai) [inline cli/src/pcluster/models/cluster.py:688]: wrong comment

File: cli/src/pcluster/models/cluster.py [hunk 3]
```diff
@@ -88,6 +90,7 @@ def __init__(
         self.throughput = Resource.init_param(throughput, default=125 if self.volume_type == "gp3" else None)
 
     def _register_validators(self):
+        # FIXME This method is not executed because subclass override the method.
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/models/cluster.py:93]: Do you mean replicate all of EbsVolumeTypeSizeValidator, EbsVolumeIposValidator, EbsVolumeThroughputValidator, EbsVolumeThroughputIopsValidator, KmsKeyValidator to subclasses? How about improving the _register_validators as we've discussed?

File: cli/src/pcluster/models/cluster.py (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/models/cluster.py:688]: Done

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 1 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/schemas/cluster_schema.py]: Done

**Review by enrico-usai — APPROVED** [review]
LGTM

**Pull request merged**